middleware/proxy/exchanger.go

```diff
@@ -13,6 +13,9 @@ type Exchanger interface {
 	Exchange(ctx context.Context, addr string, state request.Request) (*dns.Msg, error)
 	Protocol() string
 
+	// Transport returns the only transport protocol used by this Exchanger or "".
```

Comment:
But only tcp is significant? This doc string needs to be expanded so new implementers know what to do

Reply:
Now expanded.